app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,365 @@
+import android.os.Bundle;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.common.api.ResolvableApiException;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.location.LocationSettingsRequest;
+import com.google.android.gms.location.LocationSettingsResponse;
+import com.google.android.gms.location.LocationSettingsStates;
+import com.google.android.gms.location.LocationSettingsStatusCodes;
+import com.google.android.gms.tasks.Task;
+import java.lang.reflect.Method;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.wifi_hotspot.HotspotService;
+import org.kiwix.kiwixmobile.zim_manager.fileselect_view.ZimFileSelectFragment;
+
+import static org.kiwix.kiwixmobile.utils.StyleUtils.dialogStyle;
+import static org.kiwix.kiwixmobile.webserver.WebServerHelper.isStarted;
+import static org.kiwix.kiwixmobile.webserver.WebServerHelper.stopAndroidWebServer;
+import static org.kiwix.kiwixmobile.wifi_hotspot.HotspotService.checkHotspotState;
+
+public class ZimHostActivity extends AppCompatActivity implements
+    ServerStateListener {
+
+  Button startServerButton;
+  TextView serverTextView;
+
+  public static final String ACTION_TURN_ON_AFTER_O = "Turn_on_hotspot_after_oreo";
+  public static final String ACTION_TURN_OFF_AFTER_O = "Turn_off_hotspot_after_oreo";
+  private final String IP_STATE_KEY = "ip_state_key";
+  private static final int MY_PERMISSIONS_ACCESS_FINE_LOCATION = 102;
+  private Intent serviceIntent;
+  private Task<LocationSettingsResponse> task;
+  boolean flag = false;
+  String ip;
+  String TAG = ZimHostActivity.this.getClass().getSimpleName();
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_zim_host);
+
+    setUpToolbar();
+
+    startServerButton = (Button) findViewById(R.id.startServerButton);
+    serverTextView = (TextView) findViewById(R.id.server_textView);
+
+    if (savedInstanceState != null) {
+      serverTextView.setText(
+          getString(R.string.server_started_message) + " " + savedInstanceState.getString(
+              IP_STATE_KEY));
+      startServerButton.setText(getString(R.string.stop_server_label));
+    }
+
+    FragmentManager fragmentManager = getSupportFragmentManager();
+    FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
+    ZimFileSelectFragment fragment = new ZimFileSelectFragment();
+    fragmentTransaction.add(R.id.frameLayoutServer, fragment);
+    fragmentTransaction.commit();
+
+    serviceIntent = new Intent(this, HotspotService.class);
+    Context context = this;
```

Comment:
there should be no reason to do this, anywhere you use the context varaible just use `this`.

Reply:
This was fixed in a later commit.